internal/init/init.go

```diff
@@ -21,12 +21,11 @@ type initOpts struct {
 	Directory      string `opts:"help=output directory"`
 }
 
-func Register(parent opts.Opts) {
-	in := initOpts{
+func New() opts.Opts {
+	return opts.New(&initOpts{
```

Comment:
How does the subcommand get its name?

Reply:
This is tricky
```
//AddCommand to this opts instance
func (n *node) AddCommand(cmd Opts) Opts {
	sub, ok := cmd.(*node)
	if !ok {
		panic("another implementation of opts???")
	}
	//default subcommand name is struct name
	if sub.name == "" {
		sub.name = camel2dash(sub.item.val.Type().Name())
	}
	//fallback default name is package name
	pkgPath := sub.item.val.Type().PkgPath()
	if sub.name == "" && pkgPath != "" {
		parts := strings.Split(pkgPath, "/")
		sub.name = parts[len(parts)-1]
	}
	//if still no name, needs to be manually set
	if sub.name == "" {
		n.errorf("cannot add command, please set a Name()")
		return n
	}
	if _, exists := n.cmds[sub.name]; exists {
		n.errorf("cannot add command, '%s' already exists", sub.name)
		return n
	}
	n.cmds[sub.name] = sub
	sub.parent = n
	return n
}
```

Reply:
I would still personally prefer `func NewSub(sub interface{}, name string) SubOpts { ...`